src/rgw/driver/d4n/d4n_policy.h

```diff
@@ -77,32 +76,19 @@ class LFUDAPolicy : public CachePolicy {
     CacheBlock* get_victim_block(const DoutPrefixProvider* dpp, optional_yield y);
 
   public:
-    LFUDAPolicy(net::io_context& io_context, rgw::cache::CacheDriver* cacheDriver) : CachePolicy(), io(io_context), cacheDriver{cacheDriver} {
-      conn = std::make_shared<connection>(boost::asio::make_strand(io_context));
-      dir = new BlockDirectory{io};
+    LFUDAPolicy(std::shared_ptr<connection>& conn, rgw::cache::CacheDriver* cacheDriver) : CachePolicy(), 
+                                                                         conn(conn), 
+                                                                         cacheDriver(cacheDriver)
+    {
+      //conn = std::make_shared<connection>(boost::asio::make_strand(io_context));
+      dir = new BlockDirectory{conn};
```

Comment:
@samarahu : I noticed that you need to update the corresponding test files because of this change.

Reply:
Yes, thank you for reminding me. I meant to do this earlier